protocols/relay/src/priv_client/handler.rs

```diff
@@ -140,8 +151,11 @@ pub struct Handler {
     /// eventually.
     alive_lend_out_substreams: FuturesUnordered<oneshot::Receiver<void::Void>>,
 
-    circuit_deny_futs:
-        HashMap<PeerId, BoxFuture<'static, Result<(), protocol::inbound_stop::UpgradeError>>>,
+    open_circuit_futs: FuturesUnordered<
+        BoxFuture<'static, Result<inbound_stop::Circuit, inbound_stop::FatalUpgradeError>>,
+    >,
```

Comment:
It would be great if we could unify these `FuturesUnordered` somehow.

Reply:
'FuturesUnordered' is lesser evil compared to boilerplate code related to `VecDeque<SomeState>`. 
IMHO, using `OutboundOpenInfo`/`InboundOpenInfo` looks much more convenient.

Reply:
I see the additional boilerplate. Thanks for pointing it out. 

That said, I don't think `OutboundOpenInfo` and InboundOpenInfo` are worth the complexity they introduce. Thus long term I would really like to get rid of them.